File: module-1/11_Inheritance/student-exercise/java/src/main/java/com/techelevator/BankAccount.java
package com.techelevator;

import java.math.BigDecimal;

public class BankAccount {
	private String accountHolderName;
	private String accountNumber;
	private int balance;
	
	public BankAccount(String accountHolderName, String accountNumber) {
		this(accountHolderName, accountNumber, 0);
	}

	public BankAccount(String accountHolderName, String accountNumber, int balance) {
		super();
		this.balance = balance;
		this.accountHolderName = accountHolderName;
		this.accountNumber = accountNumber;
	}

	public String getAccountHolderName() {
		return accountHolderName;
	}

	public String getAccountNumber() {
		return accountNumber;
	}

	public int getBalance() {
		return balance;
	}
	
	public int deposit(int amountToDeposit) {
		balance = new BigDecimal(balance).add(new BigDecimal(amountToDeposit)).intValue();
		return balance;
	}
	
	public int withdraw(int amountToWithdraw) {
		balance = new BigDecimal(balance).subtract(new BigDecimal(amountToWithdraw)).intValue();
		return balance;
	}
}

File: module-1/11_Inheritance/exercise-final/dotnet/BankTellerExercise/Classes/BankAccount.cs
namespace BankTellerExercise.Classes
{
    public class BankAccount
    {
        private string accountHolderName;
        public string AccountHolderName
        {
            get { return accountHolderName; }
            private set { accountHolderName = value; }
        }

        private string accountNumber;
        public string AccountNumber
        {
            get { return accountNumber; }
        }

        public decimal Balance { get; private set; }


        // Constructor
        public BankAccount(string accountHolder, string accountNumber)
        {
            this.accountHolderName = accountHolder;
            this.accountNumber = accountNumber;
            this.Balance = 0M;
        }

        public BankAccount(string accountHolder, string accountNumber, decimal balance)
        {
            this.accountHolderName = accountHolder;
            this.accountNumber = accountNumber;
            this.Balance = balance;
        }

        public decimal Deposit(decimal amountToDeposit)
        {
            Balance += amountToDeposit;
            return Balance;
        }

        public virtual decimal Withdraw(decimal amountToWithdraw)
        {
            Balance -= amountToWithdraw;
            return Balance;
        }

    }
}
